### DISCUSSION TITLE
4dir vs 8dir walking -- config on entity level

### DISCUSSION THREAD

**Discussion by ok1a** [Annoraaq/grid-engine#249] [Ideas]
Hi there. Fan of your package, great work!
I was wondering, are there any plans to support directional walking configuration on an entity level?

Currently, you can set 4directional or 8directional walking as a configuration option in the global configuration. I would love to be able to define certain sprites/entities as 4 directional or 8 directional, via another configuration parameter in configuration passed to the `.addCharacter` method,  and be able to have both in the same scene.

Thanks.

**Comment by Annoraaq** [comment]
Hey @ok1a, thanks :)

I think it is a good idea. Also your [other comment](https://github.com/Annoraaq/grid-engine/issues/241#issuecomment-1075729492).

I could imagine to allow the setting in 3 places: globally, per character, per movement.

The global setting will define the default for characters. The character's setting defines the default of the movement.

That should give you maximum flexibility without forcing you to specify it all over the place.

WDYT?

**Reply by ok1a** [reply]
This sounds like the perfect way to handle this flexibly. I'd love this implemented as a feature!
